data/kotlin/data-uploader/src/main/kotlin/org/radarbase/iot/sender/KafkaAvroDataSender.kt

```diff
@@ -0,0 +1,104 @@
+package org.radarbase.iot.sender
+
+import org.apache.avro.SchemaValidationException
+import org.radarbase.config.ServerConfig
+import org.radarbase.data.RecordData
+import org.radarbase.iot.commons.auth.Authorizer
+import org.radarbase.iot.commons.util.SingletonHolder
+import org.radarbase.producer.AuthenticationException
+import org.radarbase.producer.BatchedKafkaSender
+import org.radarbase.producer.KafkaSender
+import org.radarbase.producer.rest.RestClient
+import org.radarbase.producer.rest.RestSender
+import org.radarbase.producer.rest.SchemaRetriever
+import org.radarbase.topic.AvroTopic
+import java.io.IOException
+
+open class KafkaAvroDataSender(
```

Comment:
are these `sender`s part of the Consumer on the architecture diagram

Reply:
In a way, Yes. These are like utilities that help consumers acheive what they wan. So in this case this will help the RestProxyDataConsumer to upload data to rest proxy.